### ISSUE TITLE
Release output should be clearer and more predictable

### ISSUE THREAD

**Issue by paulbellamy** [weaveworks/flux#358]
The solution to this should:
- Ideally not use the "rewriter thing" (too much), just print sequentially.
- Have clear and consistent wording
- Explain exactly what flux will do (before doing it) and as it does it
- Allow space for larger failure messages where appropriate (e.g. git clone fails)
- Be easy (or have some plan) to retrofit/upgrade it into fluxctl for users (backwards compat).

The solution to this should also solve:
- https://github.com/weaveworks/flux/issues/337 (no git error output)
- https://github.com/weaveworks/flux/issues/264 (nonsense image upgrade)
- https://github.com/weaveworks/flux/issues/219 (long output lines chomping)

**Comment by paulbellamy** [comment]
Personally, I favour something like e.g:
```
Release latest images to all services:

                Plan
                  1) Clone definition repository
                     ... OK. Cloned git@github.com:weaveworks/service-conf
                  2) Find definition files for all services
                     ... OK. Found 14 definition files.
                  3) Check registry for latest images
                     ... OK. Found 2 new images.
                  4) Update definition files to latest images
                     ... Fail: no definition files found for foobar

                Execute [Skipped: Dry-Run]
                  5) Commit and push new definitions

                  6) Rolling-update to Kubernetes

                  7) Send notifications to Slack (#cloud)

              Result: Fail: no definition files found for foobar
```
Though it wouldn't really avoid the rewriter, and not sure how you'd do it sequentially.

**Comment by squaremo** [comment]
To add or enhance those desiderata,

 - be clear about what is an action and what is a result
 - separate progress reporting ("now I'm cloning the repo ...") from results ("I updated these things"); possibly the first can go to stderr and the second to stdout, and the latter can be structured

**Comment by paulbellamy** [comment]
Fixed in #421
